app/src/main/java/com/serwylo/lexica/view/BoardView.java

```diff
@@ -59,53 +63,72 @@ private boolean isCellHighlighted(int x, int y) {
 		return false;
 	}
 
+	private final Rect textBounds = new Rect();
```

Comment:
I probably wouldn't have this as a `final` member variable, given that it still needs to be calculated at some point in the `onDraw()` method - due to a dependence on the `Paint` object. It gives a false sense that it is available to be used straight after the constructor has been run. Can it be inlined in the `onDraw()` method instead?

Reply:
I've put the Rect as a final class member to avoid continually creating an unnecessary object during onDraw, which just adds extra garbage collection. Ideally the bounds would also be calculate once and reused, but I wasn't sure if the pixel size of the boxes could change at runtime by something like rotating the device or hiding the navbar or notification bar.

Reply:
Fair enough, makes sense to me. It is quite a light weight object to create, but I do think you make a good point.